src/averageAges.js

```diff
@@ -55,7 +68,31 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const getPeople = onlyWithSon
+    ? people.filter((person) => person.sex === 'm')
+    : people;
+
+  let len = 0;
+
+  const sum = getPeople.reduce((ttl, person) => {
+    const mother = people.find((m) => m.name === person.mother);
+
+    if (mother) {
+      len++;
+
+      return ttl + person.born - mother.born;
+    }
+
+    return ttl;
+  }, 0);
+
```

Comment:
```suggestion
 const ageDif = children.map(child =>
    child.born - people.find(person => person.name === child.mother).born
  );
```

You could implement map to replace this statement and inside inmeadiately compare your element child .born  - its mother =  people.find(person => person.name === child.mother) . born

Reply:
upvote